src/planning/test/tests.cpp

```diff
@@ -80,87 +96,597 @@ void outlierCalculations(std::string filename, std::string testname) {
+  EXPECT_EQ(n1_left,  n2_left);
+  EXPECT_EQ(n1_right,  0);
+  EXPECT_EQ(n2_right,  1);
+  // track -> logCones(true);
+  // track -> logCones(false);
+}
+
+TEST(LocalPathPlanner,  distance_next_outliers) {
+  std::string filePath = "src/planning/tracks/distance_to_next.txt";
+  Track *track = new Track();
+  track->fillTrack(filePath);
+  // track -> logCones(true);
+  // track -> logCones(false);
+  int n1_left = track -> getLeftConesSize();
+  int n1_right = track -> getRightConesSize();
   track->validateCones();
-  EXPECT_EQ(track->getLeftConesSize(),
-            8);  // Both Have 9, 1 outlier should be removed
-  EXPECT_EQ(track->getRightConesSize(), 8);
+  EXPECT_FLOAT_EQ(track -> round_n(track -> getMaxDistance(true),  3),  0);
+  EXPECT_FLOAT_EQ(track -> round_n(track -> getMaxDistance(false),  3),  1.437);
+  int n2_left = track -> getLeftConesSize();
+  int n2_right = track -> getRightConesSize();
+  EXPECT_EQ(n1_left,  0);
+  EXPECT_EQ(n2_left,  1);
+  EXPECT_EQ(n1_right,  n2_right);
+  // track -> logCones(true);
+  // track -> logCones(false);
+}
+
+
+TEST(LocalPathPlanner,  path_smooth1) {
+  std::string filePath = "src/planning/tracks/path_smooth1.txt";
+  PathSmoothing *new_path = new PathSmoothing();
+  new_path->fillPath(filePath);
+  // new_path -> logPathPoints();
+  new_path -> validate(new_path -> getPath());
+  EXPECT_EQ(new_path -> getPointAmount(),  1101);
+  // new_path -> logPathPoints();
+}
+
+TEST(LocalPathPlanner,  path_smooth2) {
+  std::string filePath = "src/planning/tracks/path_smooth2.txt";
+  PathSmoothing *new_path = new PathSmoothing();
+  new_path->fillPath(filePath);
+  //new_path -> logPathPoints();
+  new_path -> validate(new_path -> getPath());
+  EXPECT_EQ(new_path -> getPointAmount(),  3801);
+  //new_path -> logPathPoints();
 }
 
-TEST(LocalPathPlanner, delauney10) {
+
+
+
+TEST(LocalPathPlanner,  delauney10) {
   std::string filePath = "src/planning/tracks/map_10.txt";
   Track *track = new Track();
   track->fillTrack(filePath);  // fill track with file data
   LocalPathPlanner *pathplanner = new LocalPathPlanner();
   std::vector<PathPoint> path;
   std::vector<PathPoint *> pathPointers = pathplanner->processNewArray(track);
   for (size_t i = 0; i < pathPointers.size(); i++) path.push_back(*pathPointers[i]);
-  std::vector<PathPoint> expected{PathPoint(1.5, 0),   PathPoint(1.75, 1),    PathPoint(2, 2),
-                                 PathPoint(2.5, 2.4), PathPoint(3, 2.8),     PathPoint(3.5, 3.15),
-                                 PathPoint(4, 3.5),   PathPoint(4.25, 4.25), PathPoint(4.5, 5)};
-  EXPECT_EQ(expected, path);
+  std::vector<PathPoint> expected{
+    PathPoint(1.5,  0.0),  PathPoint(1.75,  1.0),  PathPoint(2.25,  1.4),
+    PathPoint(2.5,  2.4),  PathPoint(3.0,  2.75),  PathPoint(3.5,  3.15),
+    PathPoint(3.75,  3.9),  PathPoint(4.25,  4.25),  PathPoint(4.5,  5.0)};
+  EXPECT_EQ(path.size(),  expected.size());
+  EXPECT_EQ(expected,  path);
+}
+
```

Comment:
Add doxygen documentation describing the scenario of the test

Reply:
✅